.github/workflows/ci.yml

```diff
@@ -264,6 +264,26 @@ jobs:
         rustup target add thumbv6m-none-eabi
         cargo check -p cordyceps --target thumbv6m-none-eabi
 
+  # run no-atomics target tests
+  maitake_sync_no_atomics_compat:
```

Comment:
If we want this job to be required for PRs to merge, we need to also add it to [the `needs:` list for the `all_systems_go` job](https://github.com/hawkw/mycelium/blob/fb9bd10d1e5dcf4c2273ff8f0f679f53d6044820/.github/workflows/ci.yml#L345-L359), which is what actually gates merging a branch. Otherwise, PRs will be able to automerge even if we've broken this check, which seems un-desirable.